lib/lib-storage/src/Upload.ts

```diff
@@ -234,6 +237,8 @@ export class Upload extends EventEmitter {
         return;
       }
 
+      this.__validateUploadPart(dataPart, this.partSize);
```

Comment:
this should be moved closer to `UploadPartCommand` to prevent any changes happening to dataPart between the validation and sending out the data